translation(JS): Web/JavaScript/Reference/Global_Objects/Object/is (#1169)
webdoky/content: files/uk/web/javascript/reference/global_objects/object/is/index.md

---
title: Object.is()
slug: Web/JavaScript/Reference/Global_Objects/Object/is
page-type: javascript-static-method
tags:
  - Comparison
  - Condition
  - Conditional
  - ECMAScript 2015
  - Equality
  - JavaScript
  - Method
  - Object
  - Polyfill
browser-compat: javascript.builtins.Object.is
---

{{JSRef}}

> Метод **`Object.is()`** визначає, чи два передані значення є насправді [одним значенням](/uk/docs/Web/JavaScript/Equality_comparisons_and_sameness#same-value_equality_using_object.is).

Reviewer: ```suggestion
Метод **`Object.is()`** визначає, чи два передані значення є насправді [одним значенням](/uk/docs/Web/JavaScript/Equality_comparisons_and_sameness#rivnist-odnoho-znachennia-za-dopomohoiu-object-is).
```

Author: який сенс давати якір на заголовок, якого не існує, якщо немає гарантії, що він залишиться актуальним після її перекладу?

Author: прийму щоб не роздувати дискусію

> Статичний метод **`Object.is()`** визначає, чи два передані значення є насправді [одним значенням](/uk/docs/Web/JavaScript/Equality_comparisons_and_sameness#rivnist-odnoho-znachennia-za-dopomohoiu-object-is).

Reviewer 2: ```suggestion
Статичний метод **`Object.is()`** визначає, чи два передані значення насправді є [одним значенням](/uk/docs/Web/JavaScript/Equality_comparisons_and_sameness#rivnist-odnoho-znachennia-za-dopomohoiu-object-is).
```

## Синтаксис

```js-nolint
Object.is(value1, value2)
```

### Параметри

- `value1`
  - : Перше значення до порівняння.
- `value2`
  - : Друге значення до порівняння.

### Повернене значення

Булеве значення, що вказує на те, чи є два переданні аргументи одним значенням, чи ні.

## Опис

> Метод `Object.is()` визначає, чи являються два передані значення [одним](/uk/docs/Web/JavaScript/Equality_comparisons_and_sameness#same-value_equality_using_object.is). Два значення вважаються одним, якщо виконується одна з наведених нижче умов:

Reviewer: ```suggestion
Метод `Object.is()` визначає, чи є два передані значення [одним і тим же значенням](/uk/docs/Web/JavaScript/Equality_comparisons_and_sameness#rivnist-odnoho-znachennia-za-dopomohoiu-object-is). Два значення вважаються одним і тим же, якщо виконується одна з наведених нижче умов:
```

> - обидва дорівнюють {{jsxref("undefined")}}

Reviewer: ```suggestion
- обидва вони – {{jsxref("undefined")}}
```

> Метод `Object.is()` визначає, чи є два передані значення [одним і тим же значенням](/uk/docs/Web/JavaScript/Equality_comparisons_and_sameness#rivnist-odnoho-znachennia-za-dopomohoiu-object-is). Два значення вважаються одним і тим же, якщо виконується одна з наведених нижче умов:

Reviewer 2: ```suggestion
Метод `Object.is()` визначає, чи є два передані значення [одним і тим же значенням](/uk/docs/Web/JavaScript/Equality_comparisons_and_sameness#rivnist-odnoho-znachennia-za-dopomohoiu-object-is). Два значення вважаються одним і тим же значенням, якщо виконується одна з наведених нижче умов:
```

> - обидва вони – {{jsxref("undefined")}}

Reviewer 2: ```suggestion
- обидва значення – {{jsxref("undefined")}}
```

> - обидва дорівнюють [`null`](/uk/docs/Web/JavaScript/Reference/Operators/null)

Reviewer: ```suggestion
- обидва вони – [`null`](/uk/docs/Web/JavaScript/Reference/Operators/null)
```

> - обидва `true` чи обидва `false`

Reviewer: ```suggestion
- обидва `true`, чи обидва `false`
```

> - обидва вони – [`null`](/uk/docs/Web/JavaScript/Reference/Operators/null)

Reviewer 2: ```suggestion
- обидва значення – [`null`](/uk/docs/Web/JavaScript/Reference/Operators/null)
```

> - обидва `true`, чи обидва `false`

Reviewer 2: ```suggestion
- обидва значення `true`, чи обидва `false`
```

> - обидва є рядками однакової довжини, з однаковими символами та в однаковій послідовності

Reviewer 2: ```suggestion
- обидва значення є рядками однакової довжини, з однаковими символами та в однаковій послідовності
```

> - обидва дорівнюють одному об'єкту (тобто обидва значення посилаються на один об'єкт в пам'яті)

Reviewer: ```suggestion
- обидва є одним і тим же об'єктом (тобто обидва значення посилаються на один об'єкт в пам'яті)
```

> - обидва є одним і тим же об'єктом (тобто обидва значення посилаються на один об'єкт в пам'яті)

Reviewer 2: ```suggestion
- обидва значення є одним і тим же об'єктом (тобто обидва значення посилаються на один об'єкт в пам'яті)
```

- обидва значення є [числами BigInt](/uk/docs/Web/JavaScript/Reference/Global_Objects/BigInt) та мають одне числове значення
- обидва значення є [символами](/uk/docs/Web/JavaScript/Reference/Global_Objects/Symbol), що посилаються на одне символьне значення
- обидва є числами та:

  - обидва дорівнюють `+0`
  - обидва дорівнюють `-0`
  - обидва дорівнюють {{jsxref("NaN")}}
  - або обидва є ненульовими значеннями, не дорівнюють {{jsxref("NaN")}}, і мають однакові числові значення

> `Object.is()` не є еквівалентом оператору [`==`](/uk/docs/Web/JavaScript/Reference/Operators/Equality). Оператор `==` застосовує різноманітні зведення до операндів з кожного боку (якщо вони не є значеннями одного типу) перед перевіркою на рівність (що призводить до такої поведінки, як істинність виразу `"" == false`), проте метод `Object.is()` не виконує зведення жодного значення.

Reviewer: ```suggestion
`Object.is()` не є еквівалентом оператору [`==`](/uk/docs/Web/JavaScript/Reference/Operators/Equality). Оператор `==` застосовує різноманітні зведення типів до операндів з обох боків (якщо вони не є значеннями одного типу) перед перевіркою на рівність (що призводить до такої поведінки, як істинність виразу `"" == false`), натомість метод `Object.is()` не виконує зведення жодного значення.
```

> `Object.is()` також _не_ еквівалентний оператору [`===`](/uk/docs/Web/JavaScript/Reference/Operators/Strict_equality). Єдиною відмінністю між `Object.is()` та `===` є їхнє ставлення до нулів зі знаками, і значень `NaN`. Оператор `===` (як і оператор `==`) сприймає числові значення `-0` та `+0` як рівні, проте сприймає значення {{jsxref("NaN")}} не рівними між собою.

Reviewer: ```suggestion
`Object.is()` також _не_ еквівалентний оператору [`===`](/uk/docs/Web/JavaScript/Reference/Operators/Strict_equality). Єдиною відмінністю між `Object.is()` та `===` є їхнє ставлення до нулів зі знаками, а також значень `NaN`. Оператор `===` (як і оператор `==`) сприймає числові значення `-0` та `+0` як рівні, проте вважає значення {{jsxref("NaN")}} не рівними між собою.
```

Author: Чим "вважає" краще за "сприймає"?
Я послуговувався тією логікою, що функція не може мати власної думки, тому щось "вважати" їй не притаманно

Reviewer: Я мав на увазі неузгодженість. Тоді пропоную такий варіант
```suggestion
`Object.is()` також _не_ еквівалентний оператору [`===`](/uk/docs/Web/JavaScript/Reference/Operators/Strict_equality). Єдиною відмінністю між `Object.is()` та `===` є їхнє ставлення до нулів зі знаками, а також значень `NaN`. Оператор `===` (як і оператор `==`) сприймає числові значення `-0` та `+0` як рівні, однак значення {{jsxref("NaN")}} – як не рівні між собою.
```

Author: ааа, точно, "сприймає... сприймає", має сенс, дякую

> `Object.is()` також _не_ еквівалентний оператору [`===`](/uk/docs/Web/JavaScript/Reference/Operators/Strict_equality). Єдиною відмінністю між `Object.is()` та `===` є їхнє ставлення до нулів зі знаками, а також значень `NaN`. Оператор `===` (як і оператор `==`) сприймає числові значення `-0` та `+0` як рівні, однак значення {{jsxref("NaN")}} – як не рівні між собою.

Reviewer 2: ```suggestion
`Object.is()` також _не_ еквівалентний оператору [`===`](/uk/docs/Web/JavaScript/Reference/Operators/Strict_equality). Єдиною відмінністю між `Object.is()` та `===` є їх ставлення до нулів зі знаками, а також значень `NaN`. Оператор `===` (як і оператор `==`) сприймає числові значення `-0` та `+0` як рівні, однак значення {{jsxref("NaN")}} – як не рівні між собою.
```
